src/test/scala/org/broadinstitute/dsde/firecloud/webservice/TrialApiServiceSpec.scala

```diff
@@ -105,99 +105,149 @@ final class TrialApiServiceSpec extends BaseServiceSpec with UserApiService with
+        s"should reject ${method.toString} method" in {
+          new RequestBuilder(method)(userAgreementPath) ~> dummyUserIdHeaders(enabledUser) ~> userServiceRoutes ~> check {
+            assert(!handled)
+            assert(localThurloeDao.agreedUsers.isEmpty)
+          }
         }
       }
-    }
 
-    allButEnabledUsers foreach { user =>
-      s"$user should not be able to agree to terms" in {
-        Put(userAgreementPath) ~> dummyUserIdHeaders(user) ~> userServiceRoutes ~> check {
-          status should equal(Forbidden)
-          assert(localThurloeDao.agreedUsers.isEmpty)
+      allButEnabledUsers foreach { user =>
+        s"should not allow a $user" in {
+          Put(userAgreementPath) ~> dummyUserIdHeaders(user) ~> userServiceRoutes ~> check {
+            status should equal(Forbidden)
+            assert(localThurloeDao.agreedUsers.isEmpty)
+          }
         }
       }
-    }
 
-    "attempting to agree to terms as enabled user" in {
-      Put(userAgreementPath) ~> dummyUserIdHeaders(enabledUser) ~> userServiceRoutes ~> check {
-        status should equal(NoContent)
-        assert(localThurloeDao.agreedUsers == Seq("enabled-user"))
+      "attempting to agree to terms as enabled user" in {
+        Put(userAgreementPath) ~> dummyUserIdHeaders(enabledUser) ~> userServiceRoutes ~> check {
+          status should equal(NoContent)
+          assert(localThurloeDao.agreedUsers == Seq("enabled-user"))
+        }
       }
     }
-  }
 
-  "Free Trial Enrollment" - {
-    val enrollPath = "/api/profile/trial"
+    val trialPathBase = "/api/profile/trial"
 
-    "User-initiated enrollment endpoint" - {
-      allHttpMethodsExcept(POST) foreach { method =>
-        s"should reject ${method.toString} method" in {
-          new RequestBuilder(method)(enrollPath) ~> dummyUserIdHeaders(enabledUser) ~> userServiceRoutes ~> check {
-            assert(!handled)
+    val enrollPaths = Seq(trialPathBase, s"$trialPathBase?operation=enroll")
+
+    enrollPaths foreach { enrollPath =>
+      s"enrollment endpoint $enrollPath" - {
+        allHttpMethodsExcept(POST) foreach { method =>
+          s"should reject ${method.toString} method" in {
+            new RequestBuilder(method)(enrollPath) ~> dummyUserIdHeaders(enabledUser) ~> userServiceRoutes ~> check {
+              assert(!handled)
+            }
           }
         }
       }
-    }
 
-    "attempting to enroll as a disabled user" - {
-      "should be a BadRequest" in {
-        Post(enrollPath) ~> dummyUserIdHeaders(disabledUser) ~> userServiceRoutes ~> check {
-          status should equal(BadRequest)
+      s"enrollment as a disabled user via $enrollPath" - {
+        "should be a BadRequest" in {
+          Post(enrollPath) ~> dummyUserIdHeaders(disabledUser) ~> userServiceRoutes ~> check {
+            status should equal(BadRequest)
+          }
+        }
+      }
+
+      s"enrollment as an enabled user that agreed to terms via $enrollPath" - {
```

Comment:
nitpick, pure language clarity, here and in the following test, this should be:
s"enrollment via $enrollPath as an enabled user that agreed to terms"
( the user enrolls via $enrollPath, the previous language makes it sound like the user agrees to terms via $enrollPath )

Reply:
Thanks; I like clarity :)